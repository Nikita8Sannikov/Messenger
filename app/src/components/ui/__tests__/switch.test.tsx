import { fireEvent, render, screen } from "@testing-library/react";

import SwitchComponent from "@components/ui/switch";

describe("SwitchComponent", () => {
	const mockOnChange = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render with initial checked state", () => {
		render(
			<SwitchComponent 
				name="test-switch"
				checked={true}
				onChange={mockOnChange}
			/>,
		);
        
		const switchElement = screen.getByRole("checkbox");
		expect(switchElement).toBeChecked();
	});

	it("should render with initial unchecked state", () => {
		render(
			<SwitchComponent 
				name="test-switch"
				checked={false}
				onChange={mockOnChange}
			/>,
		);
        
		const switchElement = screen.getByRole("checkbox");
		expect(switchElement).not.toBeChecked();
	});

	it("should call onChange when toggled", () => {
		render(
			<SwitchComponent 
				name="test-switch"
				checked={false}
				onChange={mockOnChange}
			/>,
		);
        
		const switchElement = screen.getByRole("checkbox");
		fireEvent.click(switchElement);
        
		expect(mockOnChange).toHaveBeenCalled();
	});

	it("should have correct name attribute", () => {
		render(
			<SwitchComponent 
				name="test-switch"
				checked={false}
				onChange={mockOnChange}
			/>,
		);
        
		const switchElement = screen.getByRole("checkbox");
		expect(switchElement).toHaveAttribute("name", "test-switch");
	});

	it("matches snapshot", () => {
		const { asFragment } = render(
			<SwitchComponent 
				name="test-switch"
				checked={true}
				onChange={mockOnChange}
			/>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
}); 