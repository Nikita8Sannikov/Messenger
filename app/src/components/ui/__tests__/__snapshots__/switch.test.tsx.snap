// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchComponent matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
  >
    <span
      class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
    >
      <input
        checked=""
        class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
        name="test-switch"
        type="checkbox"
      />
      <span
        class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
      />
    </span>
    <span
      class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
    />
  </span>
</DocumentFragment>
`;
